starfish/image/_single.py

```diff
@@ -0,0 +1,48 @@
+import os
+
+import collections
+from typing import Tuple
+
+from slicedimage import ImageFormat
+from slicedimage.io import resolve_url
+
+from ._base import ImageBase
+
+
+class SingleImage(ImageBase):
+    def __init__(self, data):
+        self._data = data
+
+    @classmethod
+    def from_url(cls, relativeurl: str, baseurl: str):
+        backend, name, baseurl = resolve_url(relativeurl, baseurl)
+        extension = os.path.splitext(relativeurl)[1].lstrip(".")
+        format = ImageFormat.find_by_extension(extension)
+        with backend.read_file_handle(name) as fh:
+            return SingleImage(format.reader_func(fh))
+
+    @property
+    def numpy_array(self):
+        data = self._data.view()
+        data.setflags(write=False)
+        return data
+
+    @numpy_array.setter
+    def numpy_array(self, data):
+        self._data = data.view()
+        data.setflags(write=False)
+
+    @property
+    def raw_shape(self) -> Tuple[int]:
+        return self._data.shape
+
+    @property
+    def shape(self) -> collections.OrderedDict:
+        return collections.OrderedDict()
+
+    def write(self, filepath):
+        import numpy
+        numpy.save(filepath, self._data)
+
+    def max_proj(self, *dims):
```

Comment:
I think that this should emit a warning to make users aware that they're calling a no-op function. It's pretty likely they're misunderstanding the API if they're calling this.

Reply:
I'm not sure what the right approach is for this.  I can see the POV where someone calls max_proj on an ImageBase, not knowing if it's an ImageStack or an SingleImage.  In that case, the warning is just weird and annoying.

Reply:
Isn't `ImageBase` abstract, and therefore this wouldn't happen?

Reply:
Imagine that one builds some code to load an experiment, and flatten nuclei (across Z) using max_proj.  If it happens that the experiment is 2D, then the nuclei might just be a SimpleImage, and max_proj would be a noop and also not worthy of a warning.  If the experiment is 3D, then it would be an ImageStack, and max_proj would be meaningful.

Alternatively, we could enforce ban the notion of single standalone images.